include/libint2/boys.h

```diff
@@ -1316,7 +1461,11 @@ struct TennoGmEval {
       // precision of interpolation for m=-1,0 can be insufficient, just
       // evaluate explicitly
       Real G0;
-      std::tie(Exp ? G0 : Gm_vec[0], Gmm1) = eval_G0_and_maybe_Gm1<Exp>(T, U);
+      try {
```

Comment:
should not use exceptions for non-exceptional control flow, now matter how seductive it feels. If `eval_G0_and_maybe_Gm1` can fail it needs to return std::optional<T> instead of T.